libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -250,11 +250,111 @@ const luaL_Reg AP_Logger_functions[] = {
     {NULL, NULL}
 };
 
+static int get_analog_in(lua_State *L) {
+
+    AP_HAL::AnalogSource *data = hal.analogin->channel(ANALOG_INPUT_NONE);
+
+    if (data == NULL) {
+        return 0;
```

Comment:
This should probably be an actual error, not just a silent return of nil?

Reply:
Now returns a proper error